apps/framework-cli/src/utilities/system.rs

```diff
@@ -49,3 +55,71 @@ pub async fn kill_child(child: &Child) -> Result<(), KillProcessError> {
 
     Ok(())
 }
+
+pub struct RestartingProcess {
+    monitor_task: JoinHandle<()>,
+    kill: tokio::sync::oneshot::Sender<()>,
+}
+
+pub type StartChildFn<E> = Box<dyn Fn() -> Result<Child, E> + Send + Sync>;
+
+impl RestartingProcess {
+    pub fn create<E: Debug + 'static>(start: StartChildFn<E>) -> Result<RestartingProcess, E> {
+        let child = start()?;
+        let (sender, mut receiver) = tokio::sync::oneshot::channel::<()>();
+
+        Ok(RestartingProcess {
+            monitor_task: tokio::task::spawn(async move {
+                let mut child = child;
+                loop {
+                    select! {
+                        _ = &mut receiver => {
+                            info!("Received kill signal, stopping process monitor");
+                            break;
+                        }
+                        exit_status_result = child.wait() => {
+                            match exit_status_result {
+                                Ok(exit_status) => {
+                                    if exit_status.success() {
+                                        info!("Process exited successfully");
+                                    } else {
+                                        warn!("Process exited with non-zero status: {:?}", exit_status);
+                                    }
+                                }
+                                Err(e) => {
+                                    error!("Error waiting for process: {:?}", e);
+                                }
+                            }
+                            'restart: loop {
+                                debug!("Delaying 1 second before restarting");
+                                sleep(Duration::from_secs(1)).await;
+
+                                info!("Attempting to restart process...");
```

Comment:
Could have a more precise logs? like this is the process that gets restarted. for each one of those logs